src/event/event_epoll.c

```diff
@@ -141,20 +141,20 @@ _dispatch_muxnote_create(dispatch_unote_t du, uint32_t events)
 
 	dispatch_muxnote_t dmn;
 	struct stat sb;
-	int fd = du._du->du_ident;
+	int fd = (int)du._du->du_ident;
 	int16_t filter = du._du->du_filter;
 	bool socket_listener = false;
 	sigset_t sigmask;
 
 	switch (filter) {
 	case EVFILT_SIGNAL:
-		if (!sigismember(&signals_with_unotes, du._du->du_ident)) {
+		if (!sigismember(&signals_with_unotes, (int)du._du->du_ident)) {
```

Comment:
I would take a temporary `int signo = (int)du._du->du_ident` which would make it clearer in the first place and avoid that many ugly casts